packages/next/src/client/components/react-dev-overlay/server/middleware-turbopack.ts

```diff
@@ -57,6 +60,27 @@ export async function batchedTraceSource(
   }
 }
 
+function createStackFrame(searchParams: URLSearchParams) {
+  const fileParam = searchParams.get('file')
+
+  if (!fileParam) {
+    return undefined
+  }
+
+  // rsc://React/Server/file://<filename>?42 => file://<filename>
+  const file = fileParam
+    .replace(/^rsc:\/\/React\/\w+\//, '')
```

Comment:
I guess this is safe enough but also means environment names in React must match `\w+`. Might be good to coordinate what a valid environment name is.

Reply:
We can also use `[^/]` instead of `\w` just to be safe.